modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/display/context/ClientExtensionAdminDisplayContext.java

```diff
@@ -32,12 +36,15 @@
 public class ClientExtensionAdminDisplayContext {
 
 	public ClientExtensionAdminDisplayContext(
-		CETFactory cetFactory, RenderRequest renderRequest,
-		RenderResponse renderResponse) {
+		CETFactory cetFactory, PortletRequest portletRequest,
+		PortletResponse portletResponse) {
 
 		_cetFactory = cetFactory;
-		_renderRequest = renderRequest;
-		_renderResponse = renderResponse;
+
+		_liferayPortletRequest = PortalUtil.getLiferayPortletRequest(
+			portletRequest);
+		_liferayPortletResponse = PortalUtil.getLiferayPortletResponse(
+			portletResponse);
```

Comment:
Mostly because I am ignorant to this, but why do you switch from a renderRequest to a portletRequest?

Reply:
Mmm, because I needed to access some Liferay internal methods. So, you have RenderRequest, which extends PortletRequest, in the spec. And then, there is LiferayPortletRequest which extends the spec for features that are only available in Liferay.